zypp/solver/detail/SATResolver.cc

```diff
@@ -90,6 +90,74 @@ namespace zypp
 	  }
 	}
 
+	/** Helper collecting pseudo installed items from the pool.
+	 * \todo: pseudoItems are cachable as long as pool content does not change
+	 */
+	inline sat::Queue collectPseudoInstalled( const ResPool & pool_r )
+	{
+	  sat::Queue ret;
+	  for ( const PoolItem & pi : pool_r )
+	    if ( traits::isPseudoInstalled( pi.kind() ) ) ret.push( pi.id() );
+	  return ret;
+	}
+
+	/** Copy back new \ref WeakValue to \ref PoolItem after solving.
+	 * On the fly collect orphaned items (cached by the solver for the UI)
+	 */
+	inline void solverCopyBackWeak( sat::detail::CSolver & satSolver_r, PoolItemList & orphanedItems_r )
+	{
+	  #warning CHECK CODE and assert all items weak stati are reset (resetWeak called)
```

Comment:
Is that planned to stay? Because it will always emit a warning when the file is compiled... and will get lost in all the warnings. A comment might do just as well..

Reply:
removed